pkg/util/runtimeconfig/manager_test.go

```diff
@@ -35,24 +42,42 @@ func (l *TestLimits) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return unmarshal((*plain)(l))
 }
 
-func testLoadOverrides(filename string) (interface{}, error) {
-	f, err := os.Open(filename)
-	if err != nil {
-		return nil, err
-	}
-	defer f.Close()
-
+func testLoadOverrides(r io.Reader) (interface{}, error) {
 	var overrides = &testOverrides{}
 
-	decoder := yaml.NewDecoder(f)
+	decoder := yaml.NewDecoder(r)
 	decoder.SetStrict(true)
 	if err := decoder.Decode(&overrides); err != nil {
 		return nil, err
 	}
-
 	return overrides, nil
 }
 
+func newTestOverridesManagerConfig(t *testing.T, i int32) (*atomic.Int32, ManagerConfig, func()) {
+	var config = atomic.NewInt32(i)
+
+	// create empty file
+	tempFile, err := ioutil.TempFile("", "test-validation")
+	require.NoError(t, err)
+
+	cleanup := func() {
```

Comment:
You can now register this function directly to `t.Cleanup`, no need to return it anymore. Makes code tiny bit nicer.

Cleanup function should not report errors. If this error check is part of the test (don't think that is the case here), it should not be in cleanup.